src/Config/RuntimeEntities.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.Converters;
+using Humanizer;
+
+namespace Azure.DataApiBuilder.Config;
+
+/// <summary>
+/// Represents the collection of <see cref="Entity"/> available from the RuntimeConfig.
+/// </summary>
+[JsonConverter(typeof(RuntimeEntitiesConverter))]
+public record RuntimeEntities : IEnumerable<KeyValuePair<string, Entity>>
+{
+    /// <summary>
+    /// The collection of <see cref="Entity"/> available from the RuntimeConfig.
+    /// </summary>
+    public IReadOnlyDictionary<string, Entity> Entities { get; init; }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="RuntimeEntities"/> class using a collection of entities.
+    /// 
```

Comment:
nit: remove extra line.

Reply:
I popped the extra line in there to make a clear split between the key line and the secondary information.